### PR TITLE
CONFIG:  Addition of user-secrets for obtaining the Syncfusion License

### LINKED ISSUE #77: CONFIG: Syncfusion License Key
### Safe storage of app secrets using Secret Manager in development or Azure Key Vault for production.

**Set a secret**
Run the following command from the directory in which the project file exists:

.NET CLI
`dotnet user-secrets set "Syncfusion:LicenseKey" "Your license key here"`

P.S. You can claim your community lincese here: [https://www.syncfusion.com/products/communitylicense](https://www.syncfusion.com/products/communitylicense)

**More Info:**
[Secret Manager](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-6.0&tabs=windows)
[Azure Key Vault](https://docs.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-6.0)

### PR DESCRIPTION
**Pull request by cjdutoit**
Closes #77

### CODE REVIEW

File: Taarafo.Portal.Web/Program.cs [hunk 1]
```diff
@@ -17,9 +18,24 @@ public static void Main(string[] args)
...
+                    config.AddJsonFile($"appsettings.{environment}.json", optional: true, reloadOnChange: true);
+                };
+
+                if (hostingContext.HostingEnvironment.IsDevelopment())
                 {
-                    webBuilder.UseStartup<Startup>();
-                });
+                    config.AddUserSecrets<Program>();
+                }
+            })
```
Comment (hassanhabib) [inline Taarafo.Portal.Web/Program.cs:35]: Ideally we'd be using a KeyVault to store the secrets and just bring those in.
For now, I think just adding the key in the appSettings.Development.json should be enough. Which is what you did in the code below.
Let's stick to that and remove this new configuration for app for now.

Comment (cjdutoit) [reply Taarafo.Portal.Web/Program.cs:35]: @hassanhabib yes, my thoughts was user secrets locally in development and then KeyVault for production, then the keys is not checked in.  Or are you suggesting we generate a code on Syncfusion for development and have that checked in?

Comment (hassanhabib) [reply Taarafo.Portal.Web/Program.cs:35]: @cjdutoit that's correct, you would need to generate a community license key and just keep it in the `appSettings.Development.json` file. this file is supposed to be ignored. never pushed to our master branch (or any other branch for that matter).

Comment (cjdutoit) [reply Taarafo.Portal.Web/Program.cs:35]: @hassanhabib  OK, I will revert the change.  User-secrets helpful here as its always local.

File: Taarafo.Portal.Web/Program.cs [hunk 2]
```diff
@@ -17,9 +18,19 @@ public static void Main(string[] args)
...
+                var environment = hostingContext.HostingEnvironment;
+                config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true);
+
+                if (!string.IsNullOrEmpty(environment.EnvironmentName))
                 {
-                    webBuilder.UseStartup<Startup>();
-                });
+                    config.AddJsonFile($"appsettings.{environment}.json", optional: true, reloadOnChange: true);
+                };
+            })
```
Comment (hassanhabib) [inline Taarafo.Portal.Web/Program.cs:30]: @cjdutoit one more thing here. I think ASP.NET does that outta box when you setup the environment in the `launchSettings.json` under this code:
```json
 "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }

```
If I'm not mistaken this is what you're trying to accomplish here right?

Comment (cjdutoit) [reply Taarafo.Portal.Web/Program.cs:30]: @hassanhabib did not realize that is now out of the box.  Will remove this as well.

**Pull request merged**